multi-agent/run/scripts/migrate_builtin_system.py

```
@@ -0,0 +1,445 @@
+            if key:
+                logger.info("Auto-discovered encryption key from .env file.")
+                return key
+    except Exception:
+        pass
+    return ""
+
+FERNET_VERSION_BYTE = 0x80
+FERNET_MIN_RAW_LENGTH = 73  # 1 (version) + 8 (timestamp) + 16 (IV) + 16 (min ciphertext) + 32 (HMAC)
+
+
+def run_migration(db_name: str, mongodb_ip: str, mongodb_port: str, dry_run: bool, reverse: bool = False, encryption_key: str = "") -> None:
```

Ruff (0.16.1)

[warning] Boolean-typed positional argument in function definition

(FBT001)